### PR TITLE
add support for dnx452 and preserve web.config on publish

### PR DESCRIPTION
**Pull request by anurse**
Additional changes from #2160 suggested by @davidfowl

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Unix tests fail

File: test/Bootstrapper.FunctionalTests/BootstrapperTests.cs [hunk 1]
```diff
@@ -275,6 +275,64 @@ public void BootstrapperLaunches451ClrIfDnx451IsHighestVersionInProject(string f
...
+    ""dependencies"": {
+    },
+    ""frameworks"": {
+        ""dnx452"": {},
+        ""dnx451"": {}
+    }
+}";
+            const string lockFile = @"{
+  ""locked"": false,
+  ""version"": -9996,
```
Comment (BrennanConroy) [inline test/Bootstrapper.FunctionalTests/BootstrapperTests.cs]: Isn't version 1?

Comment (anurse) [inline test/Bootstrapper.FunctionalTests/BootstrapperTests.cs]: Yep. Fixing

**Comment by anurse** [comment]
This is needed to unblock Templates, so I'm going to YOLO push it to dev now and watch the build.

**Comment by anurse** [comment]
I fixed the issue that was causing travis to fail as well.

**Pull request closed without merging**